src/ecdsa/recovery.rs

```diff
@@ -468,22 +427,20 @@ mod tests {
 }
 
```

Comment:
The following test checks the major API changes in commit `45c264e1d12b665625dbbab9e05700eca3b8b855` and verifies that error handling behavior remains consistent.

A regression could cause the new API to panic or return incorrect results instead of properly reporting errors.

```suggestion
#[test]
    fn test_recovery_error_paths() {
        // Test recovery with invalid signature
        let msg = Message::from_digest([1u8; 32]);
        let invalid_sig = RecoverableSignature::from_compact(&[0; 64], RecoveryId::Zero).unwrap();
        assert!(invalid_sig.recover_ecdsa(msg).is_err());
        
        // Test with a valid recovery ID but invalid signature data
        let result = RecoverableSignature::from_compact(&[0xFF; 63], RecoveryId::Zero); // Wrong size
        assert!(result.is_err(), "Should reject wrong-sized signature data");
    }
```